src/gui/guiFormSpecMenu.cpp

```diff
@@ -4414,7 +4369,6 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 		// If m_selected_amount has been decreased to zero, deselect
 		if (m_selected_amount == 0) {
 			m_selected_swap.clear();
-			delete m_selected_item;
 			m_selected_item = nullptr;
```

Comment:
How does this work? `ItemSpec` does not inherit from `ReferenceCounted`, so would it ever be deleted?